src/platform/user/authentication/utilities.js

```diff
@@ -41,7 +42,11 @@ export function clearRavenLoginType() {
 
 function redirect(redirectUrl, clickedEvent) {
   // Keep track of the URL to return to after auth operation.
-  sessionStorage.setItem(authnSettings.RETURN_URL, window.location.pathname);
+  sessionStorage.setItem(
+    authnSettings.RETURN_URL,
+    new URLSearchParams(window.location.search).get('next') ||
+      window.location.pathname,
```

Comment:
I think this used to be just `window.location`, which included the query, and `user-nav` would handle the next query param appropriately. But I accidentally changed this to `window.location.pathname` for some reason, which wipes the query.

Reply:
I think we need to use `window.location.href`?